src/parser.js

```diff
@@ -137,7 +133,7 @@ export function parseValue(value, type, key) {
  */
 export function parseVariablesFromLabel(kvMap, label) {
     const kv = Object.assign({}, kvMap);
-
+    const expressionRegExp = new RegExp(expressionEx, "g");
```

Comment:
This statement recompile the RegExp whenever the function `parseVariablesFromLabel` is run. :-(

Reply:
I've worry about following issue.

global flag preseve `lastIndex`.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec#Finding_successive_matches

Reply:
Good point. And this sucks.

Anyway, I've checked https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/lastIndex, and it says that `lastIndex` is reset to 0 when there is no match.

So `while (match = re.exec(...))` should do fine _unless you make an early exit_.

Reply:
Just confirmed that `lastIndex` property is writable (see the spec link in the last post).
So it is possible (and a should) to reset the `lastIndex` to zero before using the RegExp.

Test: https://jsbin.com/hopinileru/1/edit?js,console